src/components/navbar/index.tsx

```diff
@@ -238,12 +244,26 @@ const NavBar: React.FC<NavBarProps> = ({ tokens }) => {
           <Row gutter={[8, 0]}>
             {privilegeLevel !== PrivilegeLevel.NONE ? (
               <Col>
-                <Dropdown overlay={userMenu}>
+                <UserDropdown overlay={userMenu}>
                   <Button>
-                    <UserOutlined />
+                    <UserAvatar
+                      // size="large"
+                      src={
+                        asyncRequestIsComplete(contacts) &&
+                        contacts.result.mainContact.profilePicture
+                      }
+                      icon={<UserOutlined />}
+                    />
+                    <Text>
+                      {asyncRequestIsComplete(contacts)
+                        ? contacts.result.mainContact.firstName +
+                          ' ' +
+                          contacts.result.mainContact.lastName
+                        : 'Loading...'}
```

Comment:
Not too sure about how the data is stored/how the backend portion works, and what data is required and what's not, but is it safe to assume that we can access data these many levels down without an error occuring? In other words, will these be guaranteed non-null for each user?

Reply:
⛏️ template strings are nice :)